homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -34,15 +35,34 @@ async def async_setup_entry(
         name="Landis+Gyr Heat Meter",
     )
 
-    sensors = []
+    # get energy unit from config
+    if "energy_unit" in entry.data:
+        energy_unit = entry.data["energy_unit"]
+    else:
+        # get energy unit from data
+        energy_unit = energy_unit_from_data(coordinator.data)
+        if energy_unit:
+            # Add energy unit to config entry
+            new_data = {**entry.data}
+            new_data["energy_unit"] = energy_unit
+            hass.config_entries.async_update_entry(entry, data=new_data)
+            _LOGGER.info("Updated config entry with energy unit: %s", energy_unit)
```

Comment:
If the device is not available, I can still create and enable the correct entity (GJ/MWh), rather than show none or both the GJ and the MWh entity. Creating the correct entity, even if unavailable, will allow the user to see the history. If you know of another way to achieve this I can perhaps simplify this.

Reply:
> If the device is not available

If it is unavailable, the config entry should end up retrying until it succeeds. No entity should be created in the mean time. (Hence the logic above, to use the first refresh).

Reply:
Ah ok, that makes sense. Can the retry interval be tweaked? I'd hate to drain someones battery while retrying the connection. 
(This was the reason for not polling on restart in the first place, but this principle don't seem to work together with dynamically determining the unit of the device ... I've considered making it a user choise on setup, but many users won't know or don't care, so I decided to poll after all.)

Reply:
Can the retry interval be tweaked? I'd hate to drain someones battery while retrying the connection.

It is exponential, at startup. It cannot be tweaked.

Reply:
I've settled for the HA way of things. In a future PR I'll see if I can prevent draining scenario's.